File: src/code/java/hash_map/find_number_of_subarrays.java
import java.util.HashMap;
import java.util.Map;


public int fn(int[] arr, int k) {
    Map<Integer, Integer> counts = new HashMap<>();
    counts.put(0, 1);
    int ans = 0;
    int curr = 0;

    for (int num: arr) {
        // TODO: logic to change curr
        ans += counts.getOrDefault(curr - k, 0);
        counts.put(curr, counts.getOrDefault(curr, 0) + 1);
    }

    return ans;
}

File: src/code/ruby/hash_map/find_number_of_subarrays.rb
def fn(arr, k)
  counts    = Hash.new(0)
  counts[0] = 1
  ans       = 0
  curr      = 0

  arr.each do |num|
    # TODO: logic to change curr
    ans += counts[curr - k]
    counts[curr] += 1
  end

  ans
end
